ledger/src/blockstore.rs

```diff
@@ -1770,6 +1784,30 @@ impl Blockstore {
             .map(|x| x.0)
     }
 
+    fn reconcile_shreds(&self, root_slot: Slot) -> Result<()> {
+        info!("reconciling shreds from root {}", root_slot);
+        let slot_iterator = self
+            .db
+            .iter::<cf::SlotMeta>(IteratorMode::From(root_slot, IteratorDirection::Forward))?;
+        for (slot, data) in slot_iterator {
+            info!("reconciling shreds slot {}", slot);
+            let slot_meta: SlotMeta = deserialize(&data)?;
+            let current = self.find_data_indexes(slot, slot_meta.received, std::u64::MAX, &mut 0);
+            let all_shreds: Vec<Shred> = current
+                .into_iter()
+                .filter_map(|x| {
+                    let payload = self.get_data_shred(slot, x).ok()??;
+                    Shred::new_from_serialized_shred(payload).ok()
+                })
+                .collect();
+            if !all_shreds.is_empty() {
+                self.insert_shreds(all_shreds, None, false)?;
```

Comment:
hmmm I wonder how efficient this will be.  At full TPS, insert in broadcast stage can take 100-200 ms a slot. Hopefully doesn't slow down opening the ledger by too much...

Reply:
how can i check if a slot is frozen?

Reply:
usually you would check the `blockstore.is_full` function, but now without the flush, we can't be sure that even if the `SlotMeta` says the slot is full, that the underlying shreds are all going to be non-corrupted/exist on startup...